### PR TITLE
iOS Documentation Revamp

### PR DESCRIPTION
**Pull request by noobs2ninjas**
I'd like to leave this pull request open and keep adding to it and have some open communication about the updates unless someone has a better Idea of how to keep track of all the changes needing be made.

### CODE REVIEW

File: _includes/ios/local-datastore.md [hunk 1]
```diff
@@ -1,14 +1,19 @@
 # Local Datastore
 
-The Parse iOS/OS X SDK provides a local datastore which can be used to store and retrieve `PFObject`s, even when the network is unavailable. To enable this functionality, add `libsqlite3.dylib` and add `isLocalDatastoreEnabled = true` to the `ParseClientConfiguration` block used in `Parse.initialize()`.
+The Parse iOS/OS X SDK provides a local datastore which can be used to store and retrieve `PFObject`s, even when the network is unavailable. To enable this functionality add `isLocalDatastoreEnabled = true` to the `ParseClientConfiguration` block used in `Parse.initialize()` or call `Parse.enableLocalDatastore()` prior to initializing Parse.
```
Comment (TomWFox) [inline _includes/ios/local-datastore.md]: Would be good to actually provide an initialisation code sample with localdatastore enabled

File: _includes/ios/objects.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 ## The PFObject
 
-Storing data on Parse is built around the `PFObject`. Each `PFObject` contains key-value pairs of JSON-compatible data. This data is schemaless, which means that you don't need to specify ahead of time what keys exist on each `PFObject`. You simply set whatever key-value pairs you want, and our backend will store it.
+Storing data on Parse is built around the `PFObject`. Each `PFObject` contains key-value pairs of JSON-compatible data. This data is schemeless, which means that you don't need to specify ahead of time what keys exist on each `PFObject`. You simply set whatever key-value pairs you want, and our backend will store it.
```
Comment (TomWFox) [inline _includes/ios/objects.md]: I think schemaless was correct here.

File: _includes/ios/objects.md [hunk 3]
```diff
@@ -123,22 +118,35 @@ let acl = gameScore.acl
 </div>
 
 If you need to refresh an object you already have with the latest data that
-    is in the Parse Cloud, you can call the `fetch` method like so:
+is in the cloud, you can use the `fetchInBackgroundWithBlock:` or `fetchInBackgroundWithTarget:selector:` methods to provide additional logic which will run after fetching the object.
```
Comment (TomWFox) [inline _includes/ios/objects.md]:
```suggestion
is in the database, you can use the `fetchInBackgroundWithBlock:` or `fetchInBackgroundWithTarget:selector:` methods to provide additional logic which will run after fetching the object.
```

File: _includes/ios/objects.md (hunk 2 above)
Comment (noobs2ninjas) [reply _includes/ios/objects.md]: Drrp. You're right.

File: _includes/ios/files.md [hunk 4]
```diff
@@ -142,6 +142,8 @@ file?.saveInBackground({ (success: Bool, error: Error?) in
 ```
 </div>
 
-You can delete files that are referenced by objects using the [REST API]({{ site.baseUrl }}/rest/guide/#deleting-files). You will need to provide the master key in order to be allowed to delete a file.
+##Deleting Files
 
-If your files are not referenced by any object in your app, it is not possible to delete them through the REST API. You may request a cleanup of unused files in your app's Settings page. Keep in mind that doing so may break functionality which depended on accessing unreferenced files through their URL property. Files that are currently associated with an object will not be affected.
+You can delete files using the [REST API]({{site.baseUrl}}/rest/guide/#deleting-files) using the files name. You will need to provide the master key in order to be allowed to delete a file.
```
Comment (TomWFox) [inline _includes/ios/files.md]:
```suggestion
If you know the name of a file you can delete it using the [REST API]({{site.baseUrl}}/rest/guide/#deleting-files). Your master key is required for this operation.
```

File: _includes/ios/objects.md (hunk 2 above)
Comment (TomWFox) [reply _includes/ios/objects.md]: This one still needs changing

File: _includes/ios/objects.md [hunk 5]
```diff
@@ -57,9 +56,7 @@ createdAt:"2011-06-10T18:33:42Z", updatedAt:"2011-06-10T18:33:42Z"
 
 There are two things to note here. You didn't have to configure or set up a new Class called `GameScore` before running this code. Your Parse app lazily creates this Class for you when it first encounters it.
 
-There are also a few fields you don't need to specify that are provided as a convenience. `objectId` is a unique identifier for each saved object. `createdAt` and `updatedAt` represent the time that each object was created and last modified in the Parse Cloud. Each of these fields is filled in by Parse, so they don't exist on a `PFObject` until a save operation has completed.
-
-Note: You can use the `saveInBackgroundWithBlock` method to provide additional logic to run after the save completes.
+There are also a few fields you don't need to specify that are provided and set by the system as a convenience. `objectId` is a unique identifier for each saved object. `createdAt` and `updatedAt` represent the time that each object was created or last modified and saved to the Parse Server. Each of these fields is filled in by Parse, so they don't exist on a `PFObject` until the first save operation has been completed.
```
Comment (TomWFox) [inline _includes/ios/objects.md]:
```suggestion
There are also a few fields you don't need to specify that are provided and set by the system as a convenience. `objectId` is a unique identifier for each saved object. `createdAt` and `updatedAt` represent the time that each object was created or last modified and saved to the Parse Server. Each of these fields is filled in by Parse Server, so they don't exist on a `PFObject` until the first save operation has been completed.
```

File: _includes/ios/objects.md [hunk 6]
```diff
@@ -344,43 +352,38 @@ gameScore.saveInBackground()
...
-
-<div class="language-toggle" markdown="1">
-```objective_c
-[gameScore deleteInBackground];
-```
-```swift
-gameScore.deleteInBackground()
-```
-</div>
+There are a few ways to delete a  `PFObject`.  For basic asynchronous deletion of a single object call the objects `deleteInBackground` function. If you prefer to recieve a callback you can use the `deleteInBackgroundWithBlock:` or `deleteInBackgroundWithTarget:selector:` methods. If you want to block the calling thread, you can use the `delete` method. Lastly,  `deleteEventually`  network consious solution that deletes when possible but does not guarantee a timeframe for the tasks completion.
```
Comment (TomWFox) [inline _includes/ios/objects.md]:
```suggestion

There are a few ways to delete a `PFObject`. For basic asynchronous deletion of a single object call the objects `deleteInBackground` function. If you prefer to recieve a callback you can use the `deleteInBackgroundWithBlock:` or `deleteInBackgroundWithTarget:selector:` methods. If you want to block the calling thread, you can use the `delete` method. Lastly, `deleteEventually` network conscious solution that deletes when possible but does not guarantee a timeframe for the tasks completion.
```

File: _includes/ios/objects.md [hunk 7]
```diff
@@ -344,43 +352,38 @@ gameScore.saveInBackground()
...
 
-If you want to run a callback when the delete is confirmed, you can use the `deleteInBackgroundWithBlock:` or `deleteInBackgroundWithTarget:selector:` methods. If you want to block the calling thread, you can use the `delete` method.
+For deleting multiple objects use the `PFObject` static function `deleteAllInBackground` to delete an array of objects asynchronous. The same can be done while blocking the calling thread using `deleteAll`. Lastly, to recieve a callback after deleting objects asyncronously use `deleteAllInBackground:block:` as demonstrated below.
```
Comment (TomWFox) [inline _includes/ios/objects.md]:
```suggestion
For deleting multiple objects use the `PFObject` static function `deleteAllInBackground` to delete an array of objects asynchronously. The same can be done while blocking the calling thread using `deleteAll`. Lastly, to recieve a callback after deleting objects asyncronously use `deleteAllInBackground:block:` as demonstrated below.
```

Comment (noobs2ninjas) [reply _includes/ios/objects.md]: God I feel so stupid for missing stupid stuff like spelling. I guess after about 5+ hours at it last night I started getting a bit sloppy. Theres just so much that needs to be done.

**Comment by noobs2ninjas** [comment]
I'm so sorry for missing some of this dumb little things. I guess after spending 5+ hours after it last night I got a little sloppy. Theres just so much that needs to be done.

**Comment by TomWFox** [comment]
No worries, it happens to everyone, that’s what the review process is for! I’m just happy that you’re making the effort 🙂

**Comment by TomWFox** [comment]
I’ll give this a full review tomorrow.

**Comment by noobs2ninjas** [comment]
I just thought of something last night that I think needs to be addressed sooner than later. The Swift documentation has strayed quite a ways from Objective-C in terms of function names and conventions. Which was inevitable because of how different the languages are. However, all referenced function names in the documentation text is in Objective-C. My thought process is that most of those that still use Objective-C probably aren't the kind who would need to spend much time reading documentation to begin with.

My goal is to do a section or two a day till all the core documentation corrections have been completed. So, if there ever was a time to start showing preference to Swift in the documentation, now would be the time to do it. Your thoughts?

Also, would you prefer bite sized chunks to review or does submitting a lot of changes at once work for you?

**Comment by TomWFox** [comment]
I would agree on showing preference to Swift.

It's probably a bit easier if you submit separate PRs for different sections for example, however the GitHub UI makes it fairly easy to review large PRs so I'm not hugely bothered.

**Comment by noobs2ninjas** [comment]
Perfect. I end up having to rewrite most function names anyway so I'll just start doing swift. 

I can start making branches on my fork and start doing individual sections as pull requests. Also might help with consistency. 

Once we have this all done I might start go back and do the tedious task of making the Swift example default.

File: _includes/ios/objects.md [hunk 8]
```diff
@@ -69,22 +66,20 @@ Saving data to the cloud is fun, but it's even more fun to get that data out aga
...
-// immediately.  Any code that depends on the query result should be moved
-// inside the completion block above.
 ```
 ```swift
 let query = PFQuery(className:"GameScore")
-query.getObjectInBackground(withId: "xWMyZEGZ") { (gameScore: PFObject?, error: Error?) in
-    if let error = error {
-        //The query returned an error
-        print(error.localizedDescription)
+query.getObjectInBackground(withId: "xWMyZEGZ") { (object, error) in
```
Comment (TomWFox) [inline _includes/ios/objects.md]: The objective-c example still returns `gameScore` and the rest of the guide uses `gameScore` extensively so I think it is probably best to stick with that.

```suggestion
query.getObjectInBackground(withId: "xWMyZEGZ") { (gameScore, error) in
```

File: _includes/ios/objects.md [hunk 9]
```diff
@@ -123,22 +118,35 @@ let acl = gameScore.acl
 </div>
 
 If you need to refresh an object you already have with the latest data that
-    is in the Parse Cloud, you can call the `fetch` method like so:
+is in the database, you can use the `fetchInBackgroundWithBlock:` or `fetchInBackgroundWithTarget:selector:` methods to provide additional logic which will run after fetching the object.
```
Comment (TomWFox) [inline _includes/ios/objects.md]: As you've removed the `fetch` example I don't see the point in mentioning the benefit of the `withBlock` method...

```suggestion
is in the database, you can use the `fetchInBackgroundWithBlock:` or `fetchInBackgroundWithTarget:selector:` methods.
```

File: _includes/ios/objects.md [hunk 10]
```diff
@@ -344,43 +352,39 @@ gameScore.saveInBackground()
...
 
-<div class="language-toggle" markdown="1">
-```objective_c
-[gameScore deleteInBackground];
-```
-```swift
-gameScore.deleteInBackground()
-```
-</div>
+There are a few ways to delete a `PFObject`. For basic asynchronous deletion of a single object call the objects `deleteInBackground` function. If you prefer to recieve a callback you can use the `deleteInBackgroundWithBlock:` or `deleteInBackgroundWithTarget:selector:` methods. If you want to block the calling thread, you can use the `delete` method. Lastly, `deleteEventually` network conscious solution that deletes when possible but does not guarantee a timeframe for the tasks completion.
```
Comment (TomWFox) [inline _includes/ios/objects.md]:
```suggestion
There are a few ways to delete a `PFObject`. For basic asynchronous deletion of a single object call the objects `deleteInBackground` function. If you prefer to recieve a callback you can use the `deleteInBackgroundWithBlock:` or `deleteInBackgroundWithTarget:selector:` methods. If you want to block the calling thread, you can use the `delete` method. Lastly, `deleteEventually` is a network conscious option that deletes when possible but does not guarantee a timeframe for the tasks completion.
```

File: _includes/ios/objects.md [hunk 11]
```diff
@@ -344,43 +352,39 @@ gameScore.saveInBackground()
...
 
-If you want to run a callback when the delete is confirmed, you can use the `deleteInBackgroundWithBlock:` or `deleteInBackgroundWithTarget:selector:` methods. If you want to block the calling thread, you can use the `delete` method.
+For deleting multiple objects use the `PFObject` static function `deleteAllInBackground` to delete an array of objects asynchronously. The same can be done while blocking the calling thread using `deleteAll`. Lastly, to recieve a callback after deleting objects asyncronously use `deleteAllInBackground:block:` as demonstrated below.
```
Comment (TomWFox) [inline _includes/ios/objects.md:362]: Why have you completely removed reference of single object delete methods?

File: _includes/ios/objects.md [hunk 12]
```diff
@@ -461,78 +471,75 @@ You can also model a many-to-many relation using the `PFRelation` object.  This
...
-
-<div class="language-toggle" markdown="1">
-```objective_c
-[relation removeObject:post];
-```
-```swift
-relation?.remove(post)
-```
-</div>
+You can remove a post from the `PFRelation` similarly using the `removeObject:` function followed by saving the parent object.
```
Comment (TomWFox) [inline _includes/ios/objects.md:496]: Are you planning to favour the swift methods signatures as we discussed? or leaving this to another PR?

File: _includes/ios/objects.md [hunk 13]
```diff
@@ -461,78 +471,75 @@ You can also model a many-to-many relation using the `PFRelation` object.  This
...
+query.whereKey("category", equalTo: "development") 
+
+// Query first object found
+if let object = try? query.getFirstObject() {
+    // Do something with object
+} ```
 </div>
 
-For more details on `PFQuery` please look at the query portion of this guide.  A `PFRelation` behaves similar to an `NSArray` of `PFObject`, so any queries you can do on arrays of objects (other than `includeKey:`) you can do on `PFRelation`.
+You can learn more queries by visiting the [PFQuery](#queries) section.  A `PFRelation` behaves similarly to an array of  `PFObject` yet has a built in query that's capable of everything a standard `PFQuery` is other than `includeKey:`.
```
Comment (TomWFox) [inline _includes/ios/objects.md]:
```suggestion
You can learn more about queries by visiting the [PFQuery](#queries) section. A `PFRelation` behaves similarly to an array of  `PFObject` yet has a built in query that's capable of everything a standard `PFQuery` is other than `includeKey:`.
```

File: _includes/ios/objects.md [hunk 14]
```diff
@@ -688,21 +685,24 @@ If you need more complicated logic than simple property access, you can declare
...
+  
   return view
 }
 ```
 </div>
 
 ### Initializing Subclasses
 
-You should create new objects with the `object` class method. This constructs an autoreleased instance of your type and correctly handles further subclassing. To create a reference to an existing object, use `objectWithoutDataWithObjectId:`.
+You should initialize new instances of subclassses with standard initialization methods. To create a new instance to an existing Parse object, use the inherited `PFObject` class function `objectWithoutDataWithObjectId:`, or create a new object and set the objectId property manually.
```
Comment (TomWFox) [inline _includes/ios/objects.md]: Not sure this sentence makes sense...

```suggestion
You should initialize new instances of subclassses with standard initialization methods. To create a new instance of an existing Parse object, use the inherited `PFObject` class function `objectWithoutDataWithObjectId:`, or create a new object and set the objectId property manually.
```

**Review by TomWFox — CHANGES_REQUESTED** [review]
Here are some more suggested changes, sorry for the delay but It does take some concentration with detailed doc changes like this 😅

File: _includes/ios/objects.md (hunk 12 above)
Comment (noobs2ninjas) [reply _includes/ios/objects.md:496]: Unfortunately, by the time I thought about it and got your input these changes had already been committed.

File: _includes/ios/objects.md (hunk 11 above)
Comment (noobs2ninjas) [reply _includes/ios/objects.md:362]: I didnt. In fact its mentioned in the change you suggested above. Its just the way that github is the presenting the changes made.  What I did is take this:

`
>To delete an object from the cloud:

><div class="language-toggle" markdown="1">
>objective_c 
>[gameScore deleteInBackground];

>swift
>gameScore.deleteInBackground()

></div>

>If you want to run a callback when the delete is confirmed, you can use the deleteInBackgroundWithBlock: or deleteInBackgroundWithTarget:selector: methods. If you want to block the calling thread, you can use the delete method.`

and turn it into the the section you made a suggestion on above . Then I added additional information that was not documented before about PFObjects static functions for deleting arrays as you see above as well as an example for it.

Comment (TomWFox) [reply _includes/ios/objects.md:362]: Ah, my bad

File: _includes/ios/objects.md (hunk 12 above)
Comment (TomWFox) [reply _includes/ios/objects.md:496]: Ok, maybe we’ll leave this for a future pr?

Comment (noobs2ninjas) [reply _includes/ios/objects.md:496]: For sure. My plan is to kind of get through the rest of this and then return to update the docs with the swift functions as well as switch swift to the default example. Ill be doing this for the rest of the docs from here on out though.

Comment (TomWFox) [reply _includes/ios/objects.md:496]: Sounds good 👍

**Comment by TomWFox** [comment]
@noobs2ninjas this is looking good to me now, but I'd like to pull down the changes and check it all in the local host which I will do tomorrow.

@dplewis I know you're not huge on the iOS SDK but I would appreciate another set of eyes if you could give this a once over?

**Review by TomWFox — APPROVED** [review]
Sorry for the delay, I think this is good to merge.

**Comment by TomWFox** [comment]
Thanks for taking the time to improve the docs!

**Pull request merged**